pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -879,6 +884,97 @@ struct ContractionOpConversion : public OpConversionPattern<ContractionOp> {
   }
 };
 
+struct GatherOpConversion : public OpConversionPattern<GatherOp> {
+  using OpConversionPattern<GatherOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(GatherOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    IVLOG(2, "GatherOpConversion::matchAndRewrite>");
+
+    // Create an adaptor, to interpret the operands
+    GatherOpAdaptor adaptor(operands);
+
+    auto loc = op.getLoc();
+    auto ctx = rewriter.getContext();
+
+    // input values
+    auto tensor = adaptor.tensor();
+    // index values for the last dimension
+    // this is a one-dimensional array of integers
+    auto indexes = adaptor.dims();
+
+    TypeConverter typeConverter;
+    auto resultType = typeConverter.convertType(op.result().getType());
+    auto memrefType = resultType.cast<MemRefType>();
+
+    // Make an allocation for the output
+    auto resultMemRef = rewriter.create<AllocOp>(loc, memrefType).getResult();
+
+    // affine loop nest
+    // symbol representing load of index from index tensor
+
+    // we need an array of int64_t representing the results tensor's dims
+    ArrayRef<int64_t> size = memrefType.getShape();
+
+    auto loop = rewriter.create<AffineParallelOp>(
+        loc, ArrayRef<Type>{memrefType},
+        ArrayRef<AtomicRMWKind>{AtomicRMWKind::assign}, size);
+
+    auto txBuilder = loop.getBodyBuilder();
+
+    // create an affine map for loading the index, using the leading counters
```

Comment:
This block of code is correct, but it would be easier to do:
auto idxLoadMap = AffineMap::getMultiDimIdentityMap(idxDims.size(), ctx)
auto idxLoadOps = loop.getIVs().slice(0, idxDims.size())

ArrayRef and MutableArrayRef are awesome BTW (slice, etc).

Reply:
Actually you could use `loop.getIVs().take_front(idxDims.size())` I think?

Reply:
Yeah, even better.

Reply:
That's cool!